### PR TITLE
refactor: replace char_u with char 11: remove `STRLEN` part 1

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

**Comment by dundargoc** [comment]
Continuation of https://github.com/neovim/neovim/pull/20068

File: src/nvim/autocmd.c [hunk 1]
```diff
@@ -2396,20 +2396,20 @@ bool au_exists(const char *const arg) FUNC_ATTR_WARN_UNUSED_RESULT
   }
 
   // if pattern is "<buffer>", special handling is needed which uses curbuf
-  // for pattern "<buffer=N>, FNAMECMP() will work fine
+  // for pattern "<buffer=N>, path_path_fnamecmp() will work fine
```
Comment (zeertzjq) [inline src/nvim/autocmd.c:2399]: typo

File: src/nvim/autocmd.c [hunk 2]
```diff
@@ -2396,20 +2396,20 @@ bool au_exists(const char *const arg) FUNC_ATTR_WARN_UNUSED_RESULT
...
   if (pattern != NULL && STRICMP(pattern, "<buffer>") == 0) {
     buflocal_buf = curbuf;
   }
 
   // Check if there is an autocommand with the given pattern.
   for (; ap != NULL; ap = ap->next) {
     // only use a pattern when it has not been removed and has commands.
-    // For buffer-local autocommands, FNAMECMP() works fine.
+    // For buffer-local autocommands, path_path_fnamecmp() works fine.
```
Comment (zeertzjq) [inline src/nvim/autocmd.c:2407]: typo

File: src/nvim/regexp_bt.c [hunk 3]
```diff
@@ -5594,7 +5594,7 @@ static char_u *regprop(char_u *op)
   case MOPEN + 7:
   case MOPEN + 8:
   case MOPEN + 9:
-    sprintf(buf + STRLEN(buf), "MOPEN%d", OP(op) - MOPEN);
+    snprintf(buf + strlen(buf), IOSIZE, "MOPEN%d", OP(op) - MOPEN);
```
Comment (bfredl) [inline src/nvim/regexp_bt.c:5597]: 49 (50-1), not `IOSIZE`

**Comment by zeertzjq** [comment]
There seem to be more places where a wrong size is passed to `snprintf()`

**Comment by zeertzjq** [comment]
There seem to be more places where a wrong size is passed to `snprintf()`

**Comment by dundargoc** [comment]
Where? I thought fixed them all.

**Comment by zeertzjq** [comment]
Many `snprintf((char *)IObuff + strlen(IObuff), IOSIZE,` should be `snprintf(IObuff + strlen(IObuff), IOSIZE - strlen(IObuff),` instead. (The `char *` cast can be removed as well.)

**Pull request merged**